src/sysext/sysext.c

```diff
@@ -144,6 +151,89 @@ static int is_our_mount_point(const char *p) {
         return true;
 }
 
+static int need_reload(void) {
+        /* Parse the mounted images to find out if we need
+        to reload the daemon. */
+        int r;
+
+        STRV_FOREACH(p, arg_hierarchies) {
+                _cleanup_free_ char *f = NULL, *buf = NULL, *resolved = NULL, **mounted_extensions = NULL;
+
+                r = chase(*p, arg_root, CHASE_PREFIX_ROOT, &resolved, NULL);
+                if (r == -ENOENT) {
+                        log_debug_errno(r, "Hierarchy '%s%s' does not exist, ignoring.", strempty(arg_root), *p);
+                        continue;
+                }
+                if (r < 0) {
+                        log_error_errno(r, "Failed to resolve path to hierarchy '%s%s': %m, ignoring.", strempty(arg_root), *p);
+                        continue;
+                }
+
+                r = is_our_mount_point(resolved);
+                if (r < 0) {
+                        return log_oom();
+                }
+
+                if (!r)
+                        continue;
+
+                f = path_join(*p, image_class_info[arg_image_class].dot_directory_name, image_class_info[arg_image_class].short_identifier_plural);
+                if (!f) {
+                        return log_oom();
+                }
+
+                r = read_full_file(f, &buf, NULL);
+                if (r < 0) {
+                        return log_error_errno(r, "Failed to open '%s': %m", f);
+                }
+
+                mounted_extensions = strv_split_newlines(buf);
+                if (!mounted_extensions) {
+                        return log_oom();
+
+                }
+
+                STRV_FOREACH(extension, mounted_extensions) {
+                        _cleanup_strv_free_ char **extension_release = NULL;
+                        const char *extension_reload_manager = NULL;
+                        r = load_extension_release_pairs(arg_root, arg_image_class, *extension, /* relax_extension_release_check */ true, &extension_release);
+                        if (r < 0) {
+                                log_debug_errno(r, "Failed to parse extension-release metadata of %s, ignoring: %m", *extension);
+                                continue;
+                        }
+
+                        extension_reload_manager = strv_env_pairs_get(extension_release, "EXTENSION_RELOAD_MANAGER");
+                        if (!isempty(extension_reload_manager) && parse_boolean(extension_reload_manager))
```

Comment:
you are treating parsing errors from parse_boolean() as true. Which sounds wrong to me. THey should be treated as false I guess.

also, please log about parsing errors, here, in theusual `log_warning_errno(r, "Cannot parse …, ignoring: %m");` style